### PR TITLE
[Feature][Connector-V2] Socket Connector Sink

### PR DESCRIPTION
**Pull request by 531651225**
https://github.com/apache/incubator-seatunnel/issues/1946 
Add Socket Connector Sink
the output format is json format
图片

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors-v2/connector-socket/pom.xml [hunk 1]
```diff
@@ -35,6 +35,12 @@
             <artifactId>connector-common</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-format-json</artifactId>
+            <version>2.1.3-SNAPSHOT</version>
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-socket/pom.xml]: Please use `${project.version}` instead

File: seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/config/SinkConfig.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import java.io.Serializable;
+
+@Data
+public class SinkConfig implements Serializable {
+    public static final  String HOST = "host";
+    public static final  String PORT = "port";
+    private static final  String MAX_RETRIES = "max_retries";
+    private static final  int DEFAULT_MAX_RETRIES = 3;
+    private String host;
+    private Integer port;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/config/SinkConfig.java]: I think `int` is best, this only my opinion.

File: seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketSink.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+import com.google.auto.service.AutoService;
+
+import java.io.IOException;
+
+@AutoService(SeaTunnelSink.class)
+public class SocketSink extends AbstractSimpleSink<SeaTunnelRow, Void> {
+    private Config pluginConfig;
+    private SinkConfig sinkConfig;
+    private SeaTunnelRowType seaTunnelRowType;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketSink.java:43]: Please add a new line between attribute and method.

File: seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketSinkWriter.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import org.apache.seatunnel.connectors.seatunnel.common.sink.AbstractSinkWriter;
+import org.apache.seatunnel.connectors.seatunnel.socket.config.SinkConfig;
+import org.apache.seatunnel.format.json.JsonSerializationSchema;
+
+import java.io.IOException;
+
+public class SocketSinkWriter extends AbstractSinkWriter<SeaTunnelRow, Void> {
+    private final SocketClient socketClient;
+
+    protected final SerializationSchema serializationSchema;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketSinkWriter.java]: `private` is best.

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

**Comment by TyrantLucifer** [comment]
BTW, please fix CI first.

File: seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/config/SinkConfig.java [hunk 5]
```diff
@@ -25,13 +25,14 @@
...
-    private static final  String MAX_RETRIES = "max_retries";
-    private static final  int DEFAULT_MAX_RETRIES = 3;
+    public static final String HOST = "host";
+    public static final String PORT = "port";
+    private static final String MAX_RETRIES = "max_retries";
+    private static final int DEFAULT_MAX_RETRIES = 3;
     private String host;
-    private Integer port;
+    private int port;
     private Integer maxNumRetries = DEFAULT_MAX_RETRIES;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/config/SinkConfig.java]: The same as `port`

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

File: seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/config/SinkConfig.java (hunk 5 above)
Comment (531651225) [reply seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/config/SinkConfig.java]: Thank you very much for your advice. I have fixed the problems mentioned above

**Comment by TyrantLucifer** [comment]
LGTM, let's waiting CI, cc @CalvinKirs

**Comment by 531651225** [comment]
图片

The CodeQL/Analyze (Java) step in CI was cancelled,. I think the overall compilation of the project timed out.
How to solve it?

**Comment by TyrantLucifer** [comment]
> 图片
> 
> The CodeQL/Analyze (Java) step in CI was cancelled,. I think the overall compilation of the project timed out. How to solve it?

Could you please help him restart the CI process? @Hisoka-X

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**